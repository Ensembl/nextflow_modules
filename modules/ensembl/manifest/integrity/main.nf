process MANIFEST_INTEGRITY {
    tag "$meta.id"
    label 'process_low'
    container "ensemblorg/ensembl-genomio:v1.6.0"

    input:
        tuple val(meta), path(manifest_files)
    
    output:
        tuple val(meta), path("*.*", includeInputs: true), emit: all_files, optional: true
        tuple val(meta), path(integrity_file), emit: error_log, optional: true
        path "versions.yml", emit: versions

    when:
        task.ext.when == null || task.ext.when

    script:
        integrity_file = "integrity.out"
        """
        manifest_check_integrity \
            --manifest_file ./manifest.json \
            --no_fail \
            > $integrity_file
        
        # Only keep integrity file if there are errors to report
        if [ ! -s $integrity_file ]
            then rm $integrity_file
        fi


        # Get version from genomio please
        VERSION=\$(python -c "import ensembl.io.genomio; print(ensembl.io.genomio.__version__)")
        cat <<-END_VERSIONS > versions.yml
        "${task.process}":
            Genomio: \$VERSION
        END_VERSIONS
        """

    stub:
        integrity_file = "integrity.out"
        """
        echo "No change, don't create manifest error log"

        VERSION=\$(python -c "import ensembl.io.genomio; print(ensembl.io.genomio.__version__)")
        cat <<-END_VERSIONS > versions.yml
        "${task.process}":
            Genomio: \$VERSION
        END_VERSIONS
        """
}
