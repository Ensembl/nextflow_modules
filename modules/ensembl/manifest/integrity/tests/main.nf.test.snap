{
    "Integrity check stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "genes.gff3:md5,5e2fd50352e7d2ebec2bdccaa352e541",
                            "versions.yml:md5,e646dbbb63a6071c58bd4919251cd3a7"

                        ]
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    "versions.yml:md5,e646dbbb63a6071c58bd4919251cd3a7"
                ],
                "all_files": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "genes.gff3:md5,5e2fd50352e7d2ebec2bdccaa352e541",
                            "versions.yml:md5,edb15363156d984b2e23070b4e620ff4"
                        ]
                    ]
                ],
                "error_log": [
                    
                ],
                "versions": [
                    "versions.yml:md5,e646dbbb63a6071c58bd4919251cd3a7"
                ]
            }
        ],
        "timestamp": "2024-08-21T09:52:13+0000"
    }
}